### ISSUE TITLE
DataDomain - Incorrectly formatted

### ISSUE THREAD

**Issue by ctenersight** [olahallengren/sql-server-maintenance-solution#700]
https://github.com/olahallengren/sql-server-maintenance-solution/blob/9ee2513a4156a7e7252a1cb430173421b3c71053/DatabaseBackup.sql#L3710

Failing to complete backups for 50+ databases that are scheduled when writing log files.

Update line 3710:
`SET @CurrentCommand = 'declare @t table (msg nvarchar(MAX)); DECLARE @ReturnCode int; insert into @t (msg) EXECUTE @ReturnCode = dbo.emc_run_backup '''`

"Msg 0, Sev 0, State 1: Unknown token received from SQL Server [SQLSTATE HY000]
String data, right truncation [SQLSTATE 01004]
Msg 16389, Sev 16, State 1: The connection is no longer usable because the server response for a previously executed statement was incorrectly formatted. [SQLSTATE 08S01]"

https://www.dell.com/support/kbdoc/en-ca/000168243?l
[NetWorker_ MS Appagent SQL DB backups using T-SQL script fail _ Dell Canada.pdf](https://github.com/olahallengren/sql-server-maintenance-solution/files/10179968/NetWorker_.MS.Appagent.SQL.DB.backups.using.T-SQL.script.fail._.Dell.Canada.pdf)
ang=en

**Comment by eltonkerber** [comment]
[DatabaseBackup.txt](https://github.com/olahallengren/sql-server-maintenance-solution/files/12487129/DatabaseBackup.txt)

**Comment by olahallengren** [comment]
Do you know if this is still needed with the latest version of Data Domain Boost?

**Comment by olahallengren** [comment]
I am in conversation with the Data Domain team at Dell EMC about this issue.

**Comment by ctenersight** [comment]
@olahallengren I am not sure if it's still needed with the latest version, I'm currently stuck with an older Data Domain device so I'll be keeping the modification in mine.

**Comment by olahallengren** [comment]
I have suggested adding a parameter to emc_run_backup to control whether the output is returned in text format or table format.

**Comment by olahallengren** [comment]
@ctenersight and @eltonkerber, are you still using Data Domain Boost together with my scripts? Could you reach out to me over email about the issue?

**Comment by eltonkerber** [comment]
Yes, I'm still using your scripts, but I stopped using logs to file and
logs to temp table on job steps to avoid the issue.

**Comment by ctenersight** [comment]
I am but only for the next 3 weeks.

Thank you,
Cory

**Comment by olahallengren** [comment]
Thank you. I am working on a version that adds a new parameter to insert the output from dbo.emc_run_backup into a table variable.

The parameter will be:

`@DataDomainBoostNoOutputTable = 'Y'`

**Comment by olahallengren** [comment]
This has been released. If you're experiencing this issue, download and install the latest version and add `@DataDomainBoostNoOutputTable = 'Y'` to your jobs — it captures the `emc_run_backup` output into a table variable so it no longer fills the log file.
